torch/csrc/jit/tensorexpr/half_support.h

```diff
@@ -128,6 +128,46 @@ class HalfRewriter : public IRMutator {
     return v;
   }
 
+  template <typename T>
+  ExprPtr mutateArithmetic(T v) {
+    IRMutator::mutate(v);
+    if (v->dtype().scalar_type() == c10::kHalf) {
+      v->set_dtype(v->dtype().cloneWithScalarType(c10::kFloat));
+    }
+    return v;
+  }
+
+  ExprPtr mutate(AddPtr v) override {
```

Comment:
Do you mind adding a brief description of what was broken in that test and how this change fixes it?

Reply:
Sure thing, will update the summary.